### PR TITLE
[texture] seamless solar panels

### PR DESCRIPTION
**Pull request by Quezler**
> proof of concept, visually merging small solar panels into one smooth surface.

Jan-08-2020 13-55-49

> note, the corners on the "inside" aren't that perfectly smooth.

### CODE REVIEW

File: core/src/mindustry/world/blocks/power/SeamlessSolarGenerator.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            compass[i] = Core.atlas.find(name + "-" + i);
+        }
+    }
+
+    @Override
+    protected TextureRegion[] generateIcons(){
+        return new TextureRegion[]{Core.atlas.find(name + "-full")};
+    }
+
+    @Override
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/power/SeamlessSolarGenerator.java]: yes, i am aware the code below here is not dry and could be improved by using e.g. vectors/blendbits/etc 🙂

**Comment by DeltaNedas** [comment]
send pics

**Comment by CA6** [comment]
bro what are you doing with this game

**Comment by Quezler** [comment]
@CA6 nothing, i just make pr's of stuff that comes up in my mind, its up to anuke what actually makes it in 🤔

**Comment by CA6** [comment]
true but your a minor god tbh

**Comment by Quezler** [comment]
i'll take that as a compliment 🙂

**Comment by Arkanic** [comment]
what happens with large solar panels
 do they keep their little detail in the centre

**Comment by ado1928** [comment]
i dare you to do this with routers

**Comment by CA6** [comment]
seamless belts, fabrcators, junctions, wepons,....... containers and vaults would be actually nice(but snek 🐍)

**Comment by repairwa11** [comment]
这个想法很大胆，会导致一部分玩家数不清他们到底造了多少太阳能板

**Comment by Quezler** [comment]
> This idea is very bold, and will cause some players to not count how many solar panels they have made.

**Comment by sk7725** [comment]
Can you do this with the big ones? This makes the big solar panels even more useless...

**Comment by Quezler** [comment]
i suppose i could, i might revisit this pull for 6.0 🤔

**Comment by Anuken** [comment]
Implemented.

> Can you do this with the big ones?
Large solar panels are multiblocks, so it's more difficult and more visually confusing. I won't be making them seamless.

**Comment by Quezler** [comment]
lol, i did not expect this to be merged (as is) 😄

**Comment by Remint32** [comment]
Connected Textures Mod XD

**Pull request closed without merging**